index_2.js

```diff
@@ -0,0 +1,134 @@
+      $(`#${tripId}`).parent().append(tripDetails);
+    }) // .get for trip details
+
+    $(`#all-trips`).on('click', 'button:not(.details)', (event) => {
+        // stops the tripHTML click event from running
+        event.stopPropagation();
+        // have to access the class from the event with the following syntax:
+        let eventClass = event.target.attributes.class.value;
+        getForm(eventClass);
+
+        // hide the book trip button after the form appears
+        $(`#tripId`).hide();
+    }) // .click to show form
+
+
+  } // getTripData
+
+  const getForm = function getForm(id) {
+    console.log('in the getForm function');
+    // replace the button with the form to book a trip when the button is clicked
+    let idToBook = id;
+    let action = 'https://trektravel.herokuapp.com/trips/' + `${idToBook}` +  '/reservations';
+
+    let form = $(
+      `<div id="book">
+       <form action="${action}" method="post" id="book${id}" class="${id}"=>
+        <label for="name">Name:</label>
+        <input type="text" name="name"></input>
+
+        <label name="age">Age:</label>
+        <input type="text" name="age"></input>
+
+        <label for="email">Email:</label>
+        <input type="text" name="email"></input>
+
+        <input type="hidden" id="trip_id" name="trip_id" value="${id}">
+
+        <input type="submit" value="Book this trip!" class="button"></input>`
+    );
+
+    $(`.${id}`).parent().append(form)
+
+    let test = `#book${id}`;
+    debugger
+
+  } // getForm
+
+  const bookTrip = function bookTrip(id) {
+
+    console.log('in the bookTrip function!');
+    let url = $(`#books${id}`).attr('action');
+    const formData = $(`#books${id}`).serialize();
+    console.log(url, formData);
+
+    $.post(url, formData, (response) => {
+      console.log('Successful post');
+    }) // post
+
+  } // bookTrip
+
+  // click to see all the trips
+  $('#get-trips').on('click', () => {
+    $.get('https://trektravel.herokuapp.com/trips', function (data) {
+      // iterate though to add all the trips as h5 elements to the page, as well as a button to show the trip details.
+      for(i = 0; i < data.length; i ++) {
+        let id = data[i]["id"];
+        let name = data[i]["name"];
+
+        // create the jQuery object to add to the page. Give the button an ID equal to the trip id so that I can access that attribute to make an api call for the correct trip details
+        let tripHTML = $('<article class="row article">' + `<h3 large-9 column">` + name + '</h3>' + `<button id="${id}" class="details">Trip details</button>` + '</article>');
+
+        // hide the 'see all trips' button
+        $('#get-trips').hide()
+
+        // append the trip names to the all-trips section
+        $('#all-trips').append(tripHTML);
+
+        // create click event for each trip name
+        tripHTML.on('click', '.details', (event) => {
+          let tripId = $(event.target).attr('id');
+          getTripDetails(tripId);
```

Comment:
This is a good effort at refactoring (certainly better than the original), but you're still in a triply nested callback here.